ambry-store/src/main/java/com/github/ambry/store/BlobStore.java

```diff
@@ -934,6 +939,21 @@ public MessageInfo findKey(StoreKey key) throws StoreException {
     }
   }
 
+  @Override
+  public Map<StoreKey, MessageInfo> findKeys(List<? extends StoreKey> storeKeys) {
+    Map<StoreKey, MessageInfo> messageInfoMap = new HashMap<>();
+    for (StoreKey storeKey : storeKeys) {
+      try {
+        // TODO: Need to look into the possibility if its more efficient to not call findKey() multiple times in a loop,
+        // and instead search the index once for all the blobs in the list.
+        messageInfoMap.put(storeKey, findKey(storeKey));
+      } catch (StoreException stEx) {
+        logger.info("Key %s not found in findKey", storeKey.getID());
```

Comment:
not all exception means not_found from findKey, please print out the exception as well.
Also, we should use error here.

Reply:
True. I have changed the code to throw the errors from `findKey()` as is, after logging.